garak/generators/huggingface.py

```diff
@@ -159,45 +239,33 @@ def __init__(
         if not torch.cuda.is_available():
             message = "OptimumPipeline needs CUDA, but torch.cuda.is_available() returned False; quitting"
             logging.critical(message)
-            raise ValueError(message)
+            raise GarakException(message)
```

Comment:
I had this reflex also. But does it make sense for us to only ever raise `GarakException`s? `ValueError` has well-defined semantics which we don't necessarily need to duplicate, and raising GarakException instead of default exceptions losing some informativeness.

Reply:
I am inclined to use `GarakException` here because `ValueError` does not seem accurate:

>exception ValueError
Raised when an operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as [IndexError](https://docs.python.org/3/library/exceptions.html#IndexError).

This function was not passed a value with a type mismatch it inspects the environment and finds it lacking support.

Reply:
Oh, I saw the "inappropriate value" part being the `False` here. OK. Do you think `GarakException` is best-effort descriptive? If so then we should roll with that

Reply:
`GarakException` will due, I think a more clear exception type may make sense if we see available resources become an common error condition.